bcda/cli.go

```diff
@@ -201,17 +293,64 @@ func sortCCLFFiles(cclf0, cclf8, cclf9 *[]cclfFileMetadata, skipped *int) filepa
 			*skipped = *skipped + 1
 			return nil
 		}
-		if metadata.cclfNum == 0 {
-			*cclf0 = append(*cclf0, metadata)
-		} else if metadata.cclfNum == 8 {
-			*cclf8 = append(*cclf8, metadata)
-		} else if metadata.cclfNum == 9 {
-			*cclf9 = append(*cclf9, metadata)
+
+		// if we get multiple sets of files relating to the same aco for attribution purposes, concat the year
+		key := fmt.Sprintf(metadata.acoID+"_%d", metadata.timestamp.Year())
+		if (*cclfmap)[key] != nil {
+			(*cclfmap)[key] = append((*cclfmap)[key], metadata)
+		} else {
+			(*cclfmap)[key] = []cclfFileMetadata{metadata}
 		}
 		return nil
 	}
 }
 
+func validate(fileMetadata cclfFileMetadata, cclfFileValidator map[string]cclfFileValidator) error {
+	if (cclfFileMetadata{}) == fileMetadata {
+		err := errors.New("File not found")
+		log.Error(err)
+		return err
+	}
+
+	file, err := os.Open(filepath.Clean(fileMetadata.filePath))
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+
+	sc := bufio.NewScanner(file)
+	count := 0
+	var key string
+
+	if fileMetadata.cclfNum == 8 {
+		key = "CCLF8"
+	} else if fileMetadata.cclfNum == 9 {
+		key = "CCLF9"
+	}
```

Comment:
Should we return an error if it's not 8 or 9?

Reply:
we could add an error here, but my original thinking was that since the validate function only ever gets called with cclf8/cclf9 metadata then we would be okay.